// @aws-amplify/api-rest/dist/cjs/utils/serviceError.js
'use strict';

Object.defineProperty(exports, "__esModule", { value: true });
exports.parseRestApiServiceError = exports.buildRestApiServiceError = void 0;
const aws_client_utils_1 = require("@aws-amplify/core/internals/aws-client-utils");
const errors_1 = require("../errors");
/**
 * Internal-only method to create a new RestApiError from a service error.
 *
 * @internal
 */
const buildRestApiServiceError = (error) => {
    const restApiError = new errors_1.RestApiError({
        name: error?.name,
        message: error.message,
        underlyingError: error,
    });
    return restApiError;
};
exports.buildRestApiServiceError = buildRestApiServiceError;
const parseRestApiServiceError = async (response) => {
    const parsedError = await (0, aws_client_utils_1.parseJsonError)(response);
    if (!parsedError) {
        // Response is not an error.
        return;
    }
    return Object.assign((0, exports.buildRestApiServiceError)(parsedError), {
        $metadata: parsedError.$metadata,
    });
};
exports.parseRestApiServiceError = parseRestApiServiceError;
